tests/unittests/tests-hashes/tests-hashes-sha512.c

```diff
@@ -348,12 +348,12 @@ static void test_hashes_sha512_hash_update_twice(void)
     sha512_context_t sha512;
 
     sha512_init(&sha512);
-    sha512_update(&sha512, (uint8_t*)teststring, sizeof(teststring));
+    sha512_update(&sha512, (uint8_t*)teststring, strlen(teststring));
     sha512_final(&sha512, hash_update_once);
 
     sha512_init(&sha512);
     sha512_update(&sha512, (uint8_t*)teststring, 3);
-    sha512_update(&sha512, (uint8_t*)&teststring[3], sizeof(teststring)-3);
+    sha512_update(&sha512, (uint8_t*)&teststring[3], strlen(teststring)-3);
```

Comment:
since this is not related to the 64 Bit preparation -> please make this a separate PR  this should go through realy fast -- ping me if no one picks that up within a day 

i also would prefer this:
```suggestion
    sha512_update(&sha512, (uint8_t*)&teststring[3], strlen(&teststring[3]));
```